packages/ts-doctor/workspaces.js

```diff
@@ -68,8 +107,41 @@ function buildWorkspaceSources(baseDir, tsPackages) {
 
   return sources;
 }
+async function resolveTsConfig(dir, configFile = 'tsconfig.json') {
+  const tsConfig = await getTsConfig(dir, configFile);
+  // seems to mutate the input...
+  const compilerOptions = parseJsonConfigFileContent(
+    { ...(tsConfig || {}) },
+    sys,
+    dir,
+  ).options;
+
+  return { tsConfig, compilerOptions };
+}
+
+async function writeBuildConfig(dir) {
+  const configFile = 'tsconfig.build.json';
+
+  const config = await resolveTsConfig(dir, configFile);
```

Comment:
i think you just want `getTsConfig` here – you don't use the calculated `compilerOptions`